gitops/snapshot_predicate.go

```diff
@@ -23,3 +23,13 @@ func IntegrationSnapshotChangePredicate() predicate.Predicate {
 		},
 	}
 }
+
+// SnapshotTestAnnotationChangePredicate returns a predicate which filters out all objects except
+// snapshot annotation "test.appstudio.openshift.io/status" is changed for update events.
+func SnapshotTestAnnotationChangePredicate() predicate.Predicate {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
```

Comment:
I think that we need to explicitly mention the other types of events (create, delete, generic) and return false for each of them.
If I'm not mistaken, the predicate returns true by default for those events.

Reply:
Yes, you are right!

Reply:
That's right, currently the [IntegrationSnapshotEnvironmentBindingPredicate](https://github.com/redhat-appstudio/integration-service/blob/main/gitops/predicates.go#L44) makes sure that wrong events are not picked up, but it would be good to have them set up correctly.
This is out of scope for this story/PR of course, we can do it separately.